### ISSUE TITLE
Allow for contact address instead of contact person

### ISSUE THREAD

**Issue by jmaassen** [research-software-directory/RSD-as-a-service#1202]
For the NLeSC software on the RSD we are starting to have the problem that for some of the software, the contact person and/or contributors, no longer work with us. A good example is Case Lay App: 

https://research-software-directory.org/software/case-law-app

Dafne is the contact person, but left a few year ago, as did the other contributors, Vincent and Jurriaan. Therefore, all email addresses linked to this software are no longer valid. 

In some cases (like [GGIR](https://research-software-directory.org/software/ggir) or [sv-callers](https://research-software-directory.org/software/sv-callers)), the RSE continues development of the tool elsewhere or in his/her spare time. In these cases I simply update the contact person & email to reflect the new situation. 

However, there is also quite a bit of 'orphaned' software, where the RSE no longer maintains it and is not interested in being the contact person any longer. In these cases, we would like to replace the contact person by a generic "contact address" at the eScience Center (which I have already created) which we can monitor for incoming questions. Something like this: 

![Screenshot 2024-05-16 at 14-39-27 Case Law App](https://github.com/research-software-directory/RSD-as-a-service/assets/1705533/672a85a4-0377-45a3-8b8f-ce24eaaf8379)

Currently this is difficult because we always assume the contact is a person (so all the fields to fill in are person related), and this person is also automatically a contributor. 

This option would also be useable for software that has a generic email address, like the RSD itself.

**Comment by dmijatovic** [comment]
I would suggest new section "Contact" for this purpose.

**Comment by cmeessen** [comment]
I agree that it makes sense to have contact addresses in case contributors leave an organisation. Here are some thoughts:

* the user should still be able to choose from the contributors list so they do not have to enter information twice
* contact address form allows to enter general contact details like e-mail, name and/or phone number (a free text field might not be suitable because it would look different on every software page and handling for metadata could be difficult too)
* if there is no specific contact person: choose an organisation in the UI that is responsible for the software. The organisation could have a property "Default software contact" that can be set by organisation maintainers, making it easier to maintain the probably growing number of orphaned software
* be able to set multiple contacts (this was a request by our users)

**Comment by bakhshir** [comment]
The same problem with our old projects and contact for some project where people left. I really would like this solution for the RSD project pages too!

**Comment by ewan-escience** [comment]
Would it make sense to normalize the contact address? I.e., there can be one global "NLeSC software support" address that can be assigned to multiple pages. If you change something of that contact address (like email), it will be automatically be reflected on all pages? This would make for easier changing of the data globally and data aggregation, but who would be responsible for the metadata of a contact address? Someone might make a change, thinking it is for that page only, but will change it globally,

Or would it make more sense to store copies of each contact address for every software page and every project page? This allows for contact addresses to be updated on each page independently, but makes global changes and data aggregation harder. This is now also the situation for software contributors and project team members.

**Comment by cmeessen** [comment]
How would you know which global support address to show in a software or project with multiple related organisations? This would require that there is a "lead organisation", which is currently not in place. But in the long run, this might become useful. Alternatively, we could urge the maintainers to add a "backup contact organisation". Is it possible to get email addresses for organisations automatically? Having a look at ror, this doesnt seem to be a mandatory field.

**Comment by jmaassen** [comment]
In reaction to the questions above: 

- we want to allow software and projects to either have a contact person (as we have now) or a contact address (as proposed above). This is an exclusive or.
- we could rename this to "contact" or "contact information" to make it easier.
- selecting a contact person should not be harder that it currently is (selecting a contributor) - we don't want to have to fill out this information twice. 
- If needed, it's okay if the contact person information is copied into the "contact information" when you select a contributor if that is easier to implement. It should be simply overwrite this info if you select a different person.
- it's okay to manually enter the information for a contact address, as it is assumed to (almost) never change
- when a contact person is used, it should show the same information it shows now (image, name, role, organization, orcid, mail icon)
- when a contact address is used, it should show an image, some title/name, organization, mail icon. Adding stuff like phone numbers is nice, but optional for now? Maybe having a number of free text lines could work here? 
- it's fine for now if the "contact information" stays in the same place in the page. We can discuss moving it seperately.

**Comment by jmaassen** [comment]
We got an additional request from @bakhshir to support this for project pages, as many finished projects have outdated contact info.

**Comment by jmaassen** [comment]
We've had a recent addition to the RSD that adds a consortium as a contributor instead of the separate persons: 

https://research-software-directory.org/software/messy 

This is a good additional use case for contact addresses.

**Comment by dmijatovic** [comment]
@cmeessen @jmaassen @ewan-escience @dalonsoa, @mvkma I read all proposals in this issue and had look at the current code base and design. Based on all information I am proposing following approach. 

Desired functionality as I can extract it:
- Able to define contact information not directly related to contributors/team members
- Able to have more than one contact shown
- Able to use contributors, team members as well as participating organisation information as contact information
- Able to search and change contact information as RSD admin (this feature is for later)

Based on these requirements and after trying to make design I am proposing following aproach:
- Separate section on software and project page (see example below for position and design)
- Separate section on edit software and edit project page (see example below)
- One table with following columns to store all contacts: name (VARCHAR), description (VARCHAR), email (VARCHAR), phone (VARCHAR), address (JSONB), position (INT), type ('person','organisation','manual'). 
- Additional "relation" tables to keep a track where contact is used for software and project like contacts_for_software, contacts_for_projects with columns: contact_id,software_id, position. Unique key on contact_id + software_id.
- Alternatively we can create separate tables with the same structure for software and project contacts.

**This feature as proposed will be breaking change and will require migration of current contact persons into contact_info table for existing data**. At this point we choose to copy data at initial stage to contact_info table. There could be multiple entries of same contact which are related to different software and project items. The contact information would need to be updated separately. This simplifies implementation and over time we will discover additional needs of users.

## Example contacts section on software page (v1)
At the some variations of design are also shown.

<img width="1613" height="5828" alt="Image" src="https://github.com/user-attachments/assets/4c4c7f01-2ba6-431a-9358-fb9acc26b9cd" />


## Example Edit software section

<img width="1581" height="1316" alt="Image" src="https://github.com/user-attachments/assets/11bc7eac-a9fb-4ec3-a557-0a4439096fc9" />

## Example Edit contact modal

<img width="1581" height="1316" alt="Image" src="https://github.com/user-attachments/assets/47c7d624-98a3-4736-855a-ca94cd27abf6" />

## Design "variation" of software page (v2)

<img width="1613" height="5828" alt="Image" src="https://github.com/user-attachments/assets/93433fb0-7050-4c5b-b59f-59e60732de9c" />

## Design "variation" of software page (v3)

<img width="1613" height="5828" alt="Image" src="https://github.com/user-attachments/assets/99df4166-afbf-457d-8bd2-2ccf27cf656e" />

**Comment by dalonsoa** [comment]
This is a constant problem with research software, so I am very much in favour of these changes. The specific version is very much up to you. We have not launched the tool publicly, yet - hope to do it in mid/late August, so we are easy to accommodate to what suits most people.

**Comment by jmaassen** [comment]
I like the design of the contact section, as it also works well with multiple contact addresses. 

To me it makes the most sense to have the order "Contacts", "Contributors", "Organisations', and put this below the Description. The Mentions section then comes next.  

For the edit page, it could make sense to start combining these 3 in one tab? As they are all related, having 3 separate tabs with a hidden dependency may be complicated to explain to users? This would require a different design for all three, so we could also think about this in a later issue.

**Comment by dmijatovic** [comment]
@jmaassen We also have communities section which now comes after contributors, maybe move this after organisations and before mentions?
